src/libaktualizr/primary/download_nonostree_test.cc

```diff
@@ -0,0 +1,117 @@
+static std::string server = "http://127.0.0.1:";
+static std::string treehub_server = "http://127.0.0.1:";
+static boost::filesystem::path sysroot;
+
+static std::string new_rev;
+
+/*
+ * Reject non-OSTree binaries.
+ */
+TEST(Aktualizr, DownloadNonOstreeBin) {
+  TemporaryDirectory temp_dir;
+  Config conf = UptaneTestCommon::makeTestConfig(temp_dir, server);
+  conf.pacman.type = PackageManager::kOstree;
+  conf.pacman.sysroot = sysroot.string();
+  conf.pacman.ostree_server = treehub_server;
+  conf.pacman.os = "dummy-os";
+  conf.provision.device_id = "device_id";
+  conf.provision.ecu_registration_endpoint = server + "/director/ecus";
+  conf.tls.server = server;
+
+  {
+    std::shared_ptr<INvStorage> storage = INvStorage::newStorage(conf.storage);
+    OstreeManager ostree(conf.pacman, storage, nullptr, nullptr);
+
+    Aktualizr aktualizr(conf);
+
+    aktualizr.Initialize();
+
+    result::UpdateCheck update_result = aktualizr.CheckUpdates().get();
+    ASSERT_EQ(update_result.status, result::UpdateStatus::kError);
+  }
+}
+
+#ifndef __NO_MAIN__
+int main(int argc, char **argv) {
+  ::testing::InitGoogleTest(&argc, argv);
+
+  logger_init();
+
+  if (argc != 3) {
+    std::cerr << "Error: " << argv[0] << " requires the path to the aktualizr-repo utility "
+              << "and an OStree sysroot\n";
+    return EXIT_FAILURE;
+  }
+
+  Process ostree("ostree");
+  aktualizr_repo_path = argv[1];
+
+  TemporaryDirectory meta_dir;
+  TemporaryDirectory temp_sysroot;
+  sysroot = temp_sysroot / "sysroot";
+  // uses cp, as boost doesn't like to copy bad symlinks
+  int ret = system((std::string("cp -r ") + argv[2] + std::string(" ") + sysroot.string()).c_str());
+  if (ret != 0) {
+    return -1;
+  }
+  auto r = ostree.run(
+      {"rev-parse", std::string("--repo"), (sysroot / "/ostree/repo").string(), "generate-remote/generated"});
+  if (std::get<0>(r) != 0) {
+    return -1;
+  }
+
+  std::string port = TestUtils::getFreePort();
+  server += port;
+  boost::process::child http_server_process("tests/fake_http_server/fake_test_server.py", port, "-m", meta_dir.Path());
+  TestUtils::waitForServer(server + "/");
+
+  std::string treehub_port = TestUtils::getFreePort();
+  treehub_server += treehub_port;
+  TemporaryDirectory treehub_dir;
+  boost::process::child ostree_server_process("tests/sota_tools/treehub_server.py", std::string("-p"), treehub_port,
+                                              std::string("-d"), treehub_dir.PathString(), std::string("-s0.5"),
+                                              std::string("--create"));
+  TestUtils::waitForServer(treehub_server + "/");
+  r = ostree.run({"rev-parse", std::string("--repo"), treehub_dir.PathString(), "master"});
+  if (std::get<0>(r) != 0) {
+    return -1;
+  }
+  new_rev = ostree.lastStdOut();
+  boost::trim_if(new_rev, boost::is_any_of(" \t\r\n"));
+  LOG_INFO << "DEST: " << new_rev;
+
+  Process akt_repo(aktualizr_repo_path.string());
```

Comment:
Not at all critical, but I recommend renaming the "aktualizr_repo" references here to "uptane_generator". (I also changed every "akt_repo" to just "gen".)